### DISCUSSION TITLE
Maintain a list of <100% releases

### DISCUSSION THREAD

**Discussion by raccoonasdf** [beetbox/beets#5400] [Q&A]
Is there a way I can get beets to automatically keep a list of releases that are, even after import, still a sub-100% match to the release they're supposed to be? Missing tracks, track length mismatch, chromaprint mismatch: the stuff that doesn't get fixed after retagging.

The `missing` plugin helps, but those other signals are important to me, too. Like, once I looked up the fingerprint for a track on acoustid and I found out it's a different recording of the same track from a totally different release that got mixed in somehow! That type of sneaky discrepancy isn't exposed very well in the import UI (unless I'm missing something), so it'd be nice to be able to review it later.

Right now my solution is just to reimport my entire library and manually write down everything that flags `≠ tracks` to look at later, haha. It's a pretty heavy solution and a little tedious.

**Comment by Serene-Arc** [comment]
I don't believe this is currently possible with beets, especially if you write it. Basically what you're asking is to compare the old metadata with whatever is in MusicBrainz. Beets does that during the import process, but then you either accept that metadata or you use whatever the files are. We don't keep both.

You can choose not to write the new metadata to the file, and use it during a reimport to get the same matching. Put `write: no` under your import section of the config and beets will just keep a database and organise everything, but not change the files in any way.

If you'd like, you can open a feature request for us to put this into a logging system of some kind. Then beets will be able to write down a history of all the autotagging operations, which seems to be what you're looking for.

**Reply by raccoonasdf** [reply]
I'm not concerned with the old metadata. I'm talking only about when the *data* is judged to be wrong. Even with `write: yes`, an album with any of (unmatched tracks, missing tracks, wrong track length, wrong chromaprint) will never become a "100% match" without manual intervention.

My desire is to log these releases. The ones that will still be a sub-100% match on reimport, even with metadata that conforms exactly to the candidate I chose.

If I were to hack this together myself, this would do what I want:
1. import normally
2. silently try to reimport the result immediately, just to see the autotagger's output. if it isn't a 100% match, log the album and the `≠` lines in the output in a file someplace

That might be what I'll do, actually. I was just curious whether I was missing something that's already a feature.

**Reply by mkmtrading** [reply]
I’m a complete beets newbie but I was thinking the same thing, that during the import beets would annotate the percentage of the match somewhere in the metadata so that it could be queried at some point  in the future

**Reply by InvisibleFunction** [reply]
I like the idea of storing data about the files before the import. For most of my imports I don't really care if I get the release 100% correct, just saying "it's one of this release group" is close enough. Storing the original data would help me go back at a later date and pinpoint which release it is if it turns out I want to do that.
